src/runtime/ArrayBufferObject.h

```diff
@@ -53,22 +54,24 @@ class ArrayBufferObject : public Object {
     static const uint32_t maxArrayBufferSize = 210000000;
 
     void allocateBuffer(ExecutionState& state, size_t bytelength);
-    void attachBuffer(ExecutionState& state, void* buffer, size_t bytelength);
-    void attachExternalBuffer(ExecutionState& state, void* buffer, size_t bytelength);
+    void attachBuffer(BackingStore* backingStore);
     void detachArrayBuffer();
+    Optional<BackingStore*> backingStore()
+    {
+        if (m_backingStore) {
+            m_mayPointsSharedBackingStore = true;
+        }
```

Comment:
```suggestion
        if (m_backingStore) {
            m_mayPointsSharedBackingStore = true;
        }
```

By analogy to the name of this method, the user expects simple getter behavior. But it changes the status of other members here. I don't think this is a good design. AllocateBuffer and attachBuffer are the only ways to assign values to the m_backingStore. If you need to track whether the backing store is being shared, it would be better to introduce a reference count at BackingStore

Reply:
BackingStore has isShared as a member function, and I guess it's for a similar purpose.

Reply:
You are right!

Reply:
It seems to have been revived in the review, Could you tell me a reason briefly

Reply:
We need to delete BackingStore of ArrayBufferObject when detaching if buffer is not shared by other ArrayBufferObject

Reply:
Okay, some kind of optimization-based implementation.